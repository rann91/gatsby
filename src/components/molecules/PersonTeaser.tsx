/* @jsx jsx */
import GatsbyImage, { FluidObject } from 'gatsby-image'
import { Fragment } from 'react'
import { AspectRatio, Box, Heading, jsx, Text } from 'theme-ui'
import { Strapi_Person } from '../../typings/graphql'

const PersonTeaser = ({ name, job, image }: Strapi_Person) => (
  <Fragment>
    <Box>
      <AspectRatio ratio={17 / 21} bg="black">
        {image && (
          <GatsbyImage
            fluid={image.file?.childImageSharp?.fluid as FluidObject}
            alt={image.alternativeText || ''}
          />
        )}
      </AspectRatio>
    </Box>
    {name && (
      <Heading as="h3" variant="h3" pt={5} sx={{ textAlign: 'center' }}>
        {name}
      </Heading>
    )}
    {job && (
      <Text as="p" variant="p" pt={2} sx={{ textAlign: 'center' }}>
        {job}
      </Text>
    )}
  </Fragment>
)

export default PersonTeaser
