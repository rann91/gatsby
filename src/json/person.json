{
  "email": "ceo@business.com",
  "id": "5f1ae9e4fc6216c8893d8c3d",
  "image": {
    "id": "5f1ae9dafc6216c8893d8c3a",
    "alternativeText": "",
    "url": "/uploads/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg",
    "file": {
      "id": "17045070-59f1-5563-9b2b-227fa050c759",
      "publicURL": "/static/75e03c0319787e05c861002a4eab7484/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg",
      "childImageSharp": {
        "fluid": {
          "base64": "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAZABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAMEBQb/xAAWAQEBAQAAAAAAAAAAAAAAAAACAQD/2gAMAwEAAhADEAAAAYVfQ1IlLPvUkegTiv/EAB8QAAEDBAMBAAAAAAAAAAAAAAIAAQMEERIhEyIxMv/aAAgBAQABBQKQUwDjHCxjvG7WbSqOsS4QNVHii+P/xAAWEQADAAAAAAAAAAAAAAAAAAABESD/2gAIAQMBAT8BCj//xAAVEQEBAAAAAAAAAAAAAAAAAAAgIf/aAAgBAgEBPwGj/8QAHhAAAgEEAwEAAAAAAAAAAAAAAAERAhAhMRJBYXH/2gAIAQEABj8CWTTkmEJ9G0OW1kpp9tygo+X/AP/EAB0QAQACAgIDAAAAAAAAAAAAAAEAESExQWEQodH/2gAIAQEAAT8hJ+MRmAecjM1GalDvy+4JTK+4RBDhADdoLq8alUStOGvFG5rn/9oADAMBAAIAAwAAABCgHc3/xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EKPSjIj/xAAYEQEBAAMAAAAAAAAAAAAAAAABABEhMf/aAAgBAgEBPxBBsLDE9v/EACAQAQACAQQCAwAAAAAAAAAAAAEAETEhQXGhUbFhkdH/2gAIAQEAAT8Q3hNdMCZBmtAK8GI6zPCgrY0ipjQtObLXNQmUQStEJeU65gbCrpiDfrb3/YsLKH0llYgr2o3+Z0PomLk9zsT/2Q==",
          "aspectRatio": 0.8080808080808081,
          "src": "/static/75e03c0319787e05c861002a4eab7484/f422e/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg",
          "srcSet": "/static/75e03c0319787e05c861002a4eab7484/e75b5/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 160w,\n/static/75e03c0319787e05c861002a4eab7484/c01e2/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 320w,\n/static/75e03c0319787e05c861002a4eab7484/f422e/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 640w,\n/static/75e03c0319787e05c861002a4eab7484/a6352/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 960w,\n/static/75e03c0319787e05c861002a4eab7484/ec6c5/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 1280w,\n/static/75e03c0319787e05c861002a4eab7484/a6974/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 1536w",
          "sizes": "(max-width: 640px) 100vw, 640px"
        }
      }
    }
  },
  "job": "CEO",
  "name": "Michael Fassbender",
  "phone": "01 (650) 658 6823"
}
