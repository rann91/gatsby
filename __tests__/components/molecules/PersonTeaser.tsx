import { render } from '@testing-library/react'
import React from 'react'
import PersonTeaser from '../../../src/components/molecules/PersonTeaser'
import person from '../../fixtures/person-single.json'

describe('PersonTeaser', () => {
  it('renders correctly', () => {
    const { container } = render(<PersonTeaser {...person} />)

    expect(container).toMatchSnapshot()
  })
})
