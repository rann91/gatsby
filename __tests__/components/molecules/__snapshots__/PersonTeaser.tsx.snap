// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonTeaser renders correctly 1`] = `
<div>
  <div
    class="css-vurnku"
  >
    <div
      class="css-i9l3dq"
    >
      <div
        class="css-1lndym5"
      />
      <div
        class="css-1sr1905"
      >
        <div
          class=" gatsby-image-wrapper"
          style="position: relative; overflow: hidden;"
        >
          <div
            aria-hidden="true"
            style="width: 100%; padding-bottom: 123.75%;"
          />
          <img
            alt=""
            aria-hidden="true"
            src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAZABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAMEBQb/xAAWAQEBAQAAAAAAAAAAAAAAAAACAQD/2gAMAwEAAhADEAAAAYVfQ1IlLPvUkegTiv/EAB8QAAEDBAMBAAAAAAAAAAAAAAIAAQMEERIhEyIxMv/aAAgBAQABBQKQUwDjHCxjvG7WbSqOsS4QNVHii+P/xAAWEQADAAAAAAAAAAAAAAAAAAABESD/2gAIAQMBAT8BCj//xAAVEQEBAAAAAAAAAAAAAAAAAAAgIf/aAAgBAgEBPwGj/8QAHhAAAgEEAwEAAAAAAAAAAAAAAAERAhAhMRJBYXH/2gAIAQEABj8CWTTkmEJ9G0OW1kpp9tygo+X/AP/EAB0QAQACAgIDAAAAAAAAAAAAAAEAESExQWEQodH/2gAIAQEAAT8hJ+MRmAecjM1GalDvy+4JTK+4RBDhADdoLq8alUStOGvFG5rn/9oADAMBAAIAAwAAABCgHc3/xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EKPSjIj/xAAYEQEBAAMAAAAAAAAAAAAAAAABABEhMf/aAAgBAgEBPxBBsLDE9v/EACAQAQACAQQCAwAAAAAAAAAAAAEAETEhQXGhUbFhkdH/2gAIAQEAAT8Q3hNdMCZBmtAK8GI6zPCgrY0ipjQtObLXNQmUQStEJeU65gbCrpiDfrb3/YsLKH0llYgr2o3+Z0PomLk9zsT/2Q=="
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
          />
          <picture>
            <source
              sizes="(max-width: 640px) 100vw, 640px"
              srcset="/static/75e03c0319787e05c861002a4eab7484/e75b5/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 160w,
/static/75e03c0319787e05c861002a4eab7484/c01e2/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 320w,
/static/75e03c0319787e05c861002a4eab7484/f422e/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 640w,
/static/75e03c0319787e05c861002a4eab7484/a6352/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 960w,
/static/75e03c0319787e05c861002a4eab7484/ec6c5/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 1280w,
/static/75e03c0319787e05c861002a4eab7484/a6974/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 1536w"
            />
            <img
              alt=""
              loading="lazy"
              sizes="(max-width: 640px) 100vw, 640px"
              src="/static/75e03c0319787e05c861002a4eab7484/f422e/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg"
              srcset="/static/75e03c0319787e05c861002a4eab7484/e75b5/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 160w,
/static/75e03c0319787e05c861002a4eab7484/c01e2/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 320w,
/static/75e03c0319787e05c861002a4eab7484/f422e/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 640w,
/static/75e03c0319787e05c861002a4eab7484/a6352/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 960w,
/static/75e03c0319787e05c861002a4eab7484/ec6c5/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 1280w,
/static/75e03c0319787e05c861002a4eab7484/a6974/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 1536w"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;source srcset="/static/75e03c0319787e05c861002a4eab7484/e75b5/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 160w,
/static/75e03c0319787e05c861002a4eab7484/c01e2/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 320w,
/static/75e03c0319787e05c861002a4eab7484/f422e/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 640w,
/static/75e03c0319787e05c861002a4eab7484/a6352/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 960w,
/static/75e03c0319787e05c861002a4eab7484/ec6c5/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 1280w,
/static/75e03c0319787e05c861002a4eab7484/a6974/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 1536w" sizes="(max-width: 640px) 100vw, 640px" /&gt;&lt;img loading="lazy" sizes="(max-width: 640px) 100vw, 640px" srcset="/static/75e03c0319787e05c861002a4eab7484/e75b5/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 160w,
/static/75e03c0319787e05c861002a4eab7484/c01e2/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 320w,
/static/75e03c0319787e05c861002a4eab7484/f422e/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 640w,
/static/75e03c0319787e05c861002a4eab7484/a6352/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 960w,
/static/75e03c0319787e05c861002a4eab7484/ec6c5/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 1280w,
/static/75e03c0319787e05c861002a4eab7484/a6974/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg 1536w" src="/static/75e03c0319787e05c861002a4eab7484/f422e/mubariz_mehdizadeh_t3zr_Em88ehc_unsplash_4a854337d9.jpg" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
      </div>
    </div>
  </div>
  <h3
    class="css-1vsjnl7-PersonTeaser"
  >
    Michael Fassbender
  </h3>
  <p
    class="css-dt93z1-PersonTeaser"
  >
    CEO
  </p>
</div>
`;
